### DISCUSSION TITLE
Cross-server SimpleClans

### DISCUSSION THREAD

**Discussion by TomLewis** [RoinujNosde/SimpleClans#209] [Ideas]
We already have MySQL so 50% of the battle is already won.

There are quite a few mini-games that add support for SimpleClans, I REALLY want to start using these for my Mini-games server, theres just a lack of support for Cross-server SimpleClans.

What are the main reasons stopping SimpleClans being cross-server?

From a data perspective, I can see that sc_kills would need a server name/worldname at least.

Then its if an update happens on one server, like someone is kicked from a Clan for it to mirror to the other server, but that can be checked when they next login, or do as LuckPerms does and have a Bungee plugin to sit in the middle that listens for changes.

K/D can be per-server/world by just adding a new MySQL Column.

**Comment by RoinujNosde** [comment]
The main reason is caching.

I need to implement a way to communicate with the other servers about changes to update their cache.

I have plans on using the plugin messaging system. Serialize the data to json, send to other servers, deserialize.

**Comment by TomLewis** [comment]
LuckPerms is a good plugin to look at ways to communicate between servers, as well as ChatControl Red (On MC Market now) Most multi severs will use Bungee, so you have that option in the middle as a communication protocol. You could look at the Luckperms/Chatcontrol code on how they do it with a bungee plugin.

They do have options for Redis, but I don't have a Redis server so I've never used it. That would most likley be the perfect solution, but it requires people to install Redis!

My biggest concern with players trying to exploit things would be money in crew balances, if there able to figure an exploit on one server to get money into another. I would just deny the permission from Mini-games, but as "things that can go wrong" that would be my number one. That or data just getting corrupt or lost/overwritten. (I do keep backups).

Well if you need a large database to experiment with and a community to help test, PirateCraft is around! I can send you my database again if that helps.

**Comment by RoinujNosde** [comment]
Under development: #327
